ros_buildfarm/common.py

```diff
@@ -600,9 +600,21 @@ def get_packages_in_workspaces(workspace_roots, condition_context=None):
     return pkgs
 
 
-def get_xunit_publisher_types_and_patterns():
+def get_xunit_publisher_types_and_patterns(
+    ros_version, pytest_junit_compliant
+):
     types = []
-    types.append(('GoogleTestType', 'ws/test_results/**/*.xml'))
+    if ros_version == 1:
+        types.append(('GoogleTestType', 'ws/test_results/**/*.xml'))
+    elif ros_version == 2:
+        types.append(('CTestType', 'ws/test_results/*/Testing/*/Test.xml'))
+        types.append(('GoogleTestType', 'ws/test_results/**/*.gtest.xml'))
+        types.append((
+            'JUnitType' if pytest_junit_compliant else 'GoogleTestType',
```

Comment:
It's not super clear in the [pytest documentation](https://docs.pytest.org/en/latest/deprecations.html#junit-family-default-value-change-to-xunit2) or in the deprecation warnings produced, but testing locally I've observed that using pytest version >= 6 does **not** imply that the junit_family type will be xunit2. In fact, if you don't explicitly set the junit_family (either with a `pytest.ini` file or a pytest `-o` option), then we get the legacy behavior. This is what's causing CI to fail for Foxy (https://github.com/ros2/ros2/issues/1016).

I think this only happened to work for Rolling because of all of the backports explicitly setting the junit_family in pytest.ini files (https://github.com/ros2/ros2/issues/951).

Reply:
See https://github.com/ros-infrastructure/ros_buildfarm/pull/836 for a proposed fix.